ts/nni_manager/training_service/reusable/environments/localEnvironmentService.ts

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+'use strict';
+
+import * as fs from 'fs';
+import * as path from 'path';
+import * as tkill from 'tree-kill';
+import * as component from '../../../common/component';
+import { getExperimentId } from '../../../common/experimentStartupInfo';
+import { getLogger, Logger } from '../../../common/log';
+import { TrialConfigMetadataKey } from '../../common/trialConfigMetadataKey';
+import { EnvironmentInformation, EnvironmentService } from '../environment';
+import { TrialConfig } from '../../common/trialConfig';
+import { getExperimentRootDir, isAlive } from '../../../common/utils';
+import { execMkdir, runScript, execCopydir } from '../../common/util';
+
+@component.Singleton
+export class LocalEnvironmentService extends EnvironmentService {
+
+    private readonly log: Logger = getLogger();
+    private localTrialConfig: TrialConfig | undefined;
+    private experimentRootDir: string;
+    private experimentId: string;
+
+    constructor() {
+        super();
+        this.experimentId = getExperimentId();
+        this.experimentRootDir = getExperimentRootDir();
+    }
+
+    public get environmentMaintenceLoopInterval(): number {
+        return 1000;
```

Comment:
It can be faster in local env, may be `100`. It will be more efficient on fast tasks.

Reply:
If NNI refresh environment too frequently, it may cause CPU usage high. Users has reported that his CPU usage is too high when running NNI, suspect there are some un-necessary operation NNI did. Perhaps 1s for refresh environment status is enough.

Reply:
100 ms is not too high to CPU, but can save 0.9-1.8 secs to user. In my previous experience, some traditional ML tasks, takes 20 secs for each trial, 2 secs is significant gap. But it depends on how it's implement in trial dispatcher for multiple services. It may need to take slowest one.

Reply:
updated,